### PR TITLE
Add ui-react-geo package

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Add new `ui-react-geo` package for Geo related components. They will be moved out from `ui-react` in a future breaking change release.

PRs:
- https://github.com/aws-amplify/amplify-ui/pull/3694
- https://github.com/aws-amplify/amplify-ui/pull/3702
- https://github.com/aws-amplify/amplify-ui/pull/3714
- https://github.com/aws-amplify/amplify-ui/pull/3715
- https://github.com/aws-amplify/amplify-ui/pull/3728
- https://github.com/aws-amplify/amplify-ui/pull/3758
- https://github.com/aws-amplify/amplify-ui/pull/3846

#### Description of how you validated code changes
- E2E tests on example apps with MapView and LocationSearch imported from `@aws-amplify/ui-react-geo`

#### Description of how you validated workflow changes

POC PR to test workflow changes:
- https://github.com/aws-amplify/amplify-ui/pull/3716

Workflow status:
[POC / Test / Internal PRs](https://github.com/aws-amplify/amplify-ui/actions/workflows/poc-test-internal-prs.yml)
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-geo/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  collectCoverage: true,
+  collectCoverageFrom: ['<rootDir>/src/**/*.(ts|tsx)'],
+  // ignore coverage for top level "export", version and style files
+  coveragePathIgnorePatterns: ['<rootDir>/src/(index|styles|version).(ts|tsx)'],
```
Comment (calebpollman) [inline packages/react-geo/jest.config.js:5]: There's no version file right?

File: packages/react-geo/package.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "size": "yarn size-limit",
+    "test": "jest",
+    "test:ci": "yarn test && yarn check:esm",
+    "test:watch": "yarn test --watch",
+    "typecheck": "tsc --noEmit"
+  },
+  "dependencies": {
+    "@aws-amplify/ui": "5.6.0",
+    "@aws-amplify/ui-react": "4.6.0",
+    "@aws-amplify/ui-react-core": "2.1.19",
```
Comment (calebpollman) [inline packages/react-geo/package.json:45]: Are these the up to date versions?

**Review by calebpollman — COMMENTED** [review]
Looks like there is some clean up that needs to be addressed but LGTM other than that 👍🏼

File: packages/react-geo/jest.config.js (hunk 1 above)
Comment (calebpollman) [reply packages/react-geo/jest.config.js:5]: Follow up, we shouldn't need to ignore `.tsx` here either imo

File: .changeset/metal-parrots-smoke.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react": patch
```
Comment (zchenwei) [inline .changeset/metal-parrots-smoke.md:2]: I think it is worth a minor bump?

Comment (sreeramsama) [reply .changeset/metal-parrots-smoke.md:2]: Yeah, `ui-react-geo` needs to be here as a major release
Will add that

**Review by calebpollman — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]

**Pull request merged**